protocols/v2/roles-logic-sv2/src/channel_management/group_channel.rs

```diff
@@ -0,0 +1,17 @@
+use crate::channel_management::standard_channel::StandardChannel;
+use crate::mining_sv2::NewExtendedMiningJob;
+use std::collections::HashMap;
+pub struct GroupChannel {
+    channel_id: u32,
+    channels: HashMap<u32, StandardChannel>,
```

Comment:
There's no need to keep references to the actual `StandardChannel` objects. That is managed at `StandardChannelFactory` level.

A `HashSet<u32>` is sufficient here, as we only need to keep collision-free references to the "child" channel ids within this group.

```suggestion
    channel_ids: HashSet<u32>,
```